File: dotnet/Vuln/AiLegacy/AiGeneratedDao.cs
using System.Data.SQLite;
using System.Text;
using SastEval.Util;

namespace SastEval.Vuln.AiLegacy;

/// <summary>
/// CWE-89: SQL Injection via AI-generated CRUD methods using string interpolation.
/// Simulates code an AI assistant might generate without security awareness.
/// </summary>
public static class AiGeneratedDao
{
    public static IResult HandleCreate(HttpContext context)
    {
        var name = context.Request.Query["name"].ToString();
        var email = context.Request.Query["email"].ToString();

        if (string.IsNullOrEmpty(name) || string.IsNullOrEmpty(email))
        {
            return Results.BadRequest("Missing 'name' or 'email' query parameter");
        }

        var conn = DbUtil.GetConnection();

        // VULNERABLE: String interpolation in SQL (AI-generated anti-pattern)
        var sql = $"INSERT INTO users (username, email, password, role) VALUES ('{name}', '{email}', 'default', 'user')";

        using var cmd = new SQLiteCommand(sql, conn);
        cmd.ExecuteNonQuery();

        return Results.Ok($"User {name} created");
    }

    public static IResult HandleRead(HttpContext context)
    {
        var email = context.Request.Query["email"].ToString();

        if (string.IsNullOrEmpty(email))
        {
            return Results.BadRequest("Missing 'email' query parameter");
        }

        var conn = DbUtil.GetConnection();

        // VULNERABLE: String interpolation in SQL
        var sql = $"SELECT * FROM users WHERE email = '{email}'";

        using var cmd = new SQLiteCommand(sql, conn);
        using var reader = cmd.ExecuteReader();

        var sb = new StringBuilder();
        while (reader.Read())
        {
            sb.AppendLine($"User: {reader["username"]}, Email: {reader["email"]}, Role: {reader["role"]}");
        }

        return sb.Length > 0 ? Results.Text(sb.ToString()) : Results.NotFound("No user found");
    }

    public static IResult HandleUpdate(HttpContext context)
    {
        var id = context.Request.Query["id"].ToString();
        var email = context.Request.Query["email"].ToString();

        if (string.IsNullOrEmpty(id) || string.IsNullOrEmpty(email))
        {
            return Results.BadRequest("Missing 'id' or 'email' query parameter");
        }

        var conn = DbUtil.GetConnection();

        // VULNERABLE: String interpolation in SQL
        var sql = $"UPDATE users SET email = '{email}' WHERE id = {id}";

        using var cmd = new SQLiteCommand(sql, conn);
        var rows = cmd.ExecuteNonQuery();

        return Results.Ok($"Updated {rows} row(s)");
    }

    public static IResult HandleDelete(HttpContext context)
    {
        var id = context.Request.Query["id"].ToString();

        if (string.IsNullOrEmpty(id))
        {
            return Results.BadRequest("Missing 'id' query parameter");
        }

        var conn = DbUtil.GetConnection();

        // VULNERABLE: String interpolation in SQL
        var sql = $"DELETE FROM users WHERE id = {id}";

        using var cmd = new SQLiteCommand(sql, conn);
        var rows = cmd.ExecuteNonQuery();

        return Results.Ok($"Deleted {rows} row(s)");
    }
}

File: java/src/main/java/com/sasteval/vuln/ailegacy/AiGeneratedDao.java
package com.sasteval.vuln.ailegacy;

import com.sasteval.util.DbUtil;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * CWE-89: SQL Injection (AI-Generated DAO pattern)
 * VULNERABILITY: All CRUD methods use string concatenation for SQL queries.
 * Simulates typical AI-generated data-access code that lacks parameterized queries.
 */
public class AiGeneratedDao {

    public void insertUser(String name, String email) throws SQLException {
        try (Connection conn = DbUtil.getConnection();
             Statement stmt = conn.createStatement()) {
            // VULN: String concatenation in INSERT
            String sql = "INSERT INTO users (username, email) VALUES ('" + name + "', '" + email + "')";
            stmt.executeUpdate(sql);
        }
    }

    public List<Map<String, String>> getUserByName(String name) throws SQLException {
        List<Map<String, String>> results = new ArrayList<>();
        try (Connection conn = DbUtil.getConnection();
             Statement stmt = conn.createStatement()) {
            // VULN: String concatenation in SELECT
            String sql = "SELECT * FROM users WHERE username = '" + name + "'";
            ResultSet rs = stmt.executeQuery(sql);
            while (rs.next()) {
                Map<String, String> row = new HashMap<>();
                row.put("id", String.valueOf(rs.getInt("id")));
                row.put("username", rs.getString("username"));
                row.put("email", rs.getString("email"));
                results.add(row);
            }
        }
        return results;
    }

    public void updateUserEmail(String name, String email) throws SQLException {
        try (Connection conn = DbUtil.getConnection();
             Statement stmt = conn.createStatement()) {
            // VULN: String concatenation in UPDATE
            String sql = "UPDATE users SET email = '" + email + "' WHERE username = '" + name + "'";
            stmt.executeUpdate(sql);
        }
    }

    public void deleteUser(String name) throws SQLException {
        try (Connection conn = DbUtil.getConnection();
             Statement stmt = conn.createStatement()) {
            // VULN: String concatenation in DELETE
            String sql = "DELETE FROM users WHERE username = '" + name + "'";
            stmt.executeUpdate(sql);
        }
    }
}
